recipe/meta.yaml

```diff
@@ -10,33 +11,41 @@ package:
   version: {{ version }}
 
 source:
-  # we have to use git urls here to get the submodules needed for the
-  # build
-  git_url: https://github.com/dmlc/xgboost
-  git_tag: v{{ version }}
+  url: https://github.com/dmlc/xgboost/releases/download/v{{ version }}/xgboost-{{ version }}.tar.gz
+  sha256: 28bec8e821b1fefcea722d96add66024adba399063f723bc5c815f7af4a5f5e4
   patches:
     # xgboost patches
-    - 0001-Force-endian-flag-in-cross-compilation-mode.patch  # [arm64 or aarch64 or ppc64le]
-    - 0002-Enable-latest-libcxx-on-MacOS.patch  # [osx]
-    - 0003-Use-mingw-w64-path.patch
-    - 0004-Undo-dmlc-xgboost-9436.patch
+    - patches/0001-Force-endian-flag-in-cross-compilation-mode.patch  # [arm64 or aarch64 or ppc64le]
+    - patches/0002-Enable-latest-libcxx-on-MacOS.patch  # [osx]
+    - patches/0003-Use-mingw-w64-path.patch
+    - patches/0004-Undo-dmlc-xgboost-9436.patch
+    - patches/0005-CMake-Explicitly-link-with-CCCL-standalone-or-CTK.patch
+    - patches/0006-Partial-fix-for-CTK-12.5-10574.patch
```

Comment:
@hcho3 when you have a chance, could you please tidy up these patches (possibly using [`git-format-patch`]( https://git-scm.com/docs/git-format-patch ))?

As illustrated by the conversation with Jiaming above ( https://github.com/conda-forge/xgboost-feedstock/pull/167#discussion_r1690366111 ), this would likely make these now (in many cases simplified patches) easier to follow

Reply:
Done. I left the first three patches, since some of them are applied to only specific system targets.